### ISSUE TITLE
Bug: Removing external links erases everything between

### ISSUE THREAD

**Issue by b-camphart** [kevboh/longform#189]
**Versions**
Longform version: 2.0.3
Obsidian version: 1.3.7
OS [e.g. macOS, Windows, iOS, Android]: Windows
Theme: Minimal
Other plugins that you think might be relevant here:

**Describe the bug**
When two external links are in the same paragraph, the text between them, and the second link, is completely removed from the output.

**To Reproduce**
1. Create a new scene with the following content:
```markdown
[[David]] and [[Ashley]] are two wikilinks to characters.

[David](Character/Characters/David.md) and [Ashley](Character/Characters/Ashley.md) are two external links to the same characters.
```
2. Make sure the "Remove Links" step is part of the scene compilation
3. Click Compile

**Expected behavior**
The scene should be compiled to:
```markdown
David and Ashley are two wikilinks to characters.

David and Ashley are two external links to the same characters.
```

**Screenshots**
![Capture](https://github.com/kevboh/longform/assets/50385408/bb41d5f4-0c18-4e03-ab1b-690bd200a03b)

**Additional context**

**Comment by kevboh** [comment]
Regex leak! Should be a simple fix. Thanks.

**Comment by b-camphart** [comment]
@kevboh this was automatically closed due to #191 being merged, but it wasn't actually resolved.  191 will just make it easier to fix
